subprojects/gradle/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/suite/TestSuiteProvider.kt

```diff
@@ -36,16 +40,28 @@ internal interface TestSuiteProvider {
             )
 
             if (reportSkippedTests) {
-                report.sendSkippedTests(
-                    skippedTests = suite.skippedTests
-                        // do not report skip here, to prevent final test status rewrite (green from last run - ok)
-                        .filter { (_, verdict) ->
-                            verdict !is Excluded.BySignatures || verdict.source != Signatures.Source.PreviousRun
-                        }
-                        .map { (targetTest, verdict) ->
-                            targetTest.test to verdict.reason
-                        }
-                )
+                val skippedTests = suite.skippedTests
+                    // do not report skip here, to prevent final test status rewrite (green from last run - ok)
+                    .filter { (_, verdict) ->
+                        verdict !is Excluded.BySignatures || verdict.source != Signatures.Source.PreviousRun
+                    }
+                    .map { (targetTest, verdict) ->
+                        targetTest.test to verdict.reason
+                    }
+
+                if (useInMemoryReport) {
```

Comment:
Can we hide this decision inside the report?

It's easy to forget about checking toggle before sending tests.

Reply:
we can't; it uses separate interfaces depends on flag